### ISSUE TITLE
Separate Hook Timeout

### ISSUE THREAD

**Issue by BenHesketh21** [helm/helm#11983]
I am using pre install hooks to set an environment for the chart and post-delete hooks to clean up, it is working really well but one question I had was is there a possibility of having a separate timeout for hooks as to the chart in general? Or even the concept of an install timeout and an upgrade timeout?

I could just set the timeout really high but I like having them at a sensible amount for the upgrades to find issues quickly but that means that the pre install hooks take too long and the install times out. Just wondered if this had discussed before or there's a better way for me to do this.

Thank you for you time

**Comment by joejulian** [comment]
No, there is not a way to specify a separate timeout. I could see that it might be a useful idea, but that would require someone to be willing and able to contribute this as a feature.

**Comment by BenHesketh21** [comment]
I still believe this would be useful I have made a PR above.
